Framework/Core/src/Dispatcher.cxx

```diff
@@ -48,9 +49,16 @@ void Dispatcher::init(InitContext& ctx)
 {
   LOG(DEBUG) << "Reading Data Sampling Policies...";
 
-  std::unique_ptr<ConfigurationInterface> cfg = ConfigurationFactory::getConfiguration(mReconfigurationSource);
-  auto policiesTree = cfg->getRecursive("dataSamplingPolicies");
-  mPolicies.clear();
+  boost::property_tree::ptree policiesTree;
+
+  if (mReconfigurationSource.empty() == false) {
+    std::unique_ptr<ConfigurationInterface> cfg = ConfigurationFactory::getConfiguration(mReconfigurationSource);
+    policiesTree = cfg->getRecursive("dataSamplingPolicies");
+    mPolicies.clear();
+  } else {
+    policiesTree = ctx.options().get<boost::property_tree::ptree>("qc-config-ptree");
```

Comment:
Could you rename `"qc-config-ptree"`to `"sampling-config-ptree"`? In principle we try to keep Data Sampling decoupled from QC.